core/src/main/scala/kafka/controller/ReplicaStateMachine.scala

```diff
@@ -300,7 +300,7 @@ class ReplicaStateMachine(controller: KafkaController) extends Logging {
   }
 
   def replicasInDeletionStates(topic: String): Set[PartitionAndReplica] = {
-    val deletionStates = Set(ReplicaDeletionStarted, ReplicaDeletionSuccessful, ReplicaDeletionIneligible)
+    val deletionStates = Set[ReplicaState](ReplicaDeletionStarted, ReplicaDeletionSuccessful, ReplicaDeletionIneligible)
```

Comment:
The values are all objects so they end up as `ReplicaState with Product with Serializable` so the `contains(r._2)` fails to typecheck.